File: update_langs.js
const fs = require('fs');
const path = require('path');
console.log('Starting...');
const locDir = path.join('c:/react_projects/ezan-app/src/localization');
const files = fs.readdirSync(locDir).filter(f => f.endsWith('.json'));

const baseEn = {
  verseBody: 'Click to read the verse of the day!',
  hadithBody: 'Click to read the hadith of the day!',
  quoteBody: 'Click to read the quote of the day!'
};

const baseTr = {
  verseBody: 'Günün ayetini okumak için týklayýn!',
  hadithBody: 'Günün hadisini okumak için týklayýn!',
  quoteBody: 'Günün sözünü okumak için týklayýn!'
};

async function translateText(text, targetLang) {
  try {
    const res = await fetch('https://translate.googleapis.com/translate_a/single?client=gtx&sl=en&tl=' + targetLang + '&dt=t&q=' + encodeURIComponent(text));
    const json = await res.json();
    return json[0].map(item => item[0]).join('');
  } catch (e) {
    console.error('Translation failed for', targetLang, text);
    return text;
  }
}

async function run() {
  for (const file of files) {
    const lang = file.split('.')[0];
    const filePath = path.join(locDir, file);
    const data = JSON.parse(fs.readFileSync(filePath, 'utf8'));
    
    if (!data.daily) data.daily = {};

    if (lang === 'en') {
      data.daily.verseBody = baseEn.verseBody;
      data.daily.hadithBody = baseEn.hadithBody;
      data.daily.quoteBody = baseEn.quoteBody;
    } else if (lang === 'tr') {
      data.daily.verseBody = baseTr.verseBody;
      data.daily.hadithBody = baseTr.hadithBody;
      data.daily.quoteBody = baseTr.quoteBody;
    } else {
      console.log('Translating for', lang);
      data.daily.verseBody = await translateText(baseEn.verseBody, lang);
      data.daily.hadithBody = await translateText(baseEn.hadithBody, lang);
      data.daily.quoteBody = await translateText(baseEn.quoteBody, lang);
    }

    fs.writeFileSync(filePath, JSON.stringify(data, null, 2));
    console.log('Updated', file);
  }
}

run().catch(console.error);

File: update_langs.py
import os
import json
import urllib.request
import urllib.parse

loc_dir = r"c:\react_projects\ezan-app\src\localization"

base_en = {
  "verseBody": "Click to read the verse of the day!",
  "hadithBody": "Click to read the hadith of the day!",
  "quoteBody": "Click to read the quote of the day!"
}

base_tr = {
  "verseBody": "Günün ayetini okumak için tıklayın!",
  "hadithBody": "Günün hadisini okumak için tıklayın!",
  "quoteBody": "Günün sözünü okumak için tıklayın!"
}

def translate(text, target):
    url = f"https://translate.googleapis.com/translate_a/single?client=gtx&sl=en&tl={target}&dt=t&q={urllib.parse.quote(text)}"
    req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
    try:
        with urllib.request.urlopen(req) as response:
            data = json.loads(response.read().decode('utf-8'))
            return ''.join([part[0] for part in data[0]])
    except Exception as e:
        print(f"Error translating to {target}: {e}")
        return text

for file_name in os.listdir(loc_dir):
    if file_name.endswith('.json'):
        lang = file_name.split('.')[0]
        file_path = os.path.join(loc_dir, file_name)
        with open(file_path, 'r', encoding='utf-8') as f:
            data = json.load(f)
            
        if 'daily' not in data:
            data['daily'] = {}
            
        if lang == 'en':
            data['daily'].update(base_en)
        elif lang == 'tr':
            data['daily'].update(base_tr)
        else:
            print(f"Translating for {lang}...")
            data['daily']['verseBody'] = translate(base_en['verseBody'], lang)
            data['daily']['hadithBody'] = translate(base_en['hadithBody'], lang)
            data['daily']['quoteBody'] = translate(base_en['quoteBody'], lang)
            
        with open(file_path, 'w', encoding='utf-8') as f:
            json.dump(data, f, ensure_ascii=False, indent=2)
        print(f"Updated {file_name}")
